src/lib/components/Error/WidgetPropValidator.tsx

```diff
@@ -0,0 +1,49 @@
+import { DefaultAddress } from 'lib/components/Swap'
+import { WidgetProps } from 'lib/components/Widget'
+import { IntegrationError } from 'lib/errors'
+import { PropsWithChildren, useEffect, useMemo, useState } from 'react'
+
+import { isAddress } from '../../../utils'
+
+function isAddressOrAddressMap(addressOrMap: DefaultAddress): boolean {
+  if (typeof addressOrMap === 'object') {
+    return Object.values(addressOrMap).every((address) => isAddress(address))
+  }
+  if (typeof addressOrMap === 'string') {
+    return typeof isAddress(addressOrMap) === 'string'
```

Comment:
weird syntax, but `isAddress` returns an address or `false`

Reply:
instead of using a typeof check, return `Boolean(isAddress(addressOrMap)`. More readable imo